src/shipment/controller.js

```diff
@@ -505,3 +510,323 @@ async function getPopulatedShipmentData(shipmentId = undefined) {
+    table,
+    layout: "noBorders",
+    style: { fillColor: "#e1e1e1" },
+    // margin: [0, 15, 0, 15]
+  };
+
+  return ret;
+}
+
+
+
+function _pdf_makePackingBlock(customerTitle, shippingContact) {
+  const bypassShipToCheck =
+    !process.env.SHOPQ_URL && process.env.NODE_ENV === "DEBUG";
+
+  if (!shippingContact && !bypassShipToCheck) {
+    return HTTPError(
+      "Shipping contact not set! Please contact sales rep.",
+      400
+    );
+  }
+
+  const body = [[{ text: "SHIP TO", bold: true }]];
+
+  const { address, name } = shippingContact;
+  const { line1, line2, line3, line4 } = address || {};
+
+  body.push([customerTitle], [line1]);
+
+  if (line2) body.push([line2]);
+  if (line3) body.push([line3]);
+  if (line4) body.push([line4]);
+  body.push(["ATTN: " + name]);
+
+  const ret = {
+    table: { body },
+    layout: "noBorders",
+    margin: [0, 20, 0, 20],
+  };
+
+  return ret;
+}
+
+/**
+ * Make signature block
+ * @param {String} packedByuUsernam
+ */
+ function _pdf_makeSignaturesBlock(deliveryMethod) {
+
+  const _table =  { 
+    table: {
+      widths: ["*", "*"],
+    },
+    unbreakable: true,
+  };
+
+  let deliveryTypeString;
+  let secondColumnSignoff;
+  if ( ['DROPOFF', 'PICKUP'].includes(deliveryMethod) ) {
+    deliveryTypeString = deliveryMethod[0] + deliveryMethod.slice(1).toLowerCase() + ' by:';
+    secondColumnSignoff = "X __________________________";
+  }
+  else {
+    deliveryTypeString = '';
+    secondColumnSignoff = '';
+  }
+
+  _table.table.body = [
+    [
+      {
+        text: "Packed by: ",
+        bold: true,
+        border: [false, false, false, false],
+      },
+      {
+        text: deliveryTypeString,
+        bold: true,
+        border: [false, false, false, false],
+        alignment: "left",
+      },
+    ],
+    [
+      {
+        text: "X __________________________",
+        border: [false, false, false, false],
+      },
+      {
+        text: secondColumnSignoff,
+        border: [false, false, false, false],
+      },
+    ],
+  ];
+
+  return _table;
+}
+
+/**
+ * Make the manifest block.
+ * This includes only the line items in the
+ * @param {any[]} items Items in the packing slip
+ */
+ function _pdf_makeManifestBlock(items, tableTitleArr, pageBreak) {
+  const body = [
+    [
+      {
+        colSpan: 2,
+        text: tableTitleArr[0],
+        fillColor: "#cccccc",
+        bold: true,
+        border: [true, true, false, true],
+      },
+      {},
+      {
+        colSpan: 2,
+        text: tableTitleArr[1],
+        fillColor: "#cccccc",
+        bold: true,
+        border: [false, true, true, true],
+        alignment: "right",
+      },
+      {},
+    ],
+    [
+      { text: "LINE" },
+      { text: "ITEM" },
+      { text: "ORDER QTY" },
+      { text: "SHIP QTY" },
+    ],
+  ];
+
+  body[1].forEach((x) => {
+    x.fillColor = "#cccccc";
+    x.bold = true;
+    x.alignment = "center";
+  });
+
+  let lineNumber = 0;
+  for (let i of items) {
```

Comment:
Don't use `let` in a `for of` loop. It implies `i` will be manipulated, which is strongly discouraged.

Reply:
Understood